src/test/java/spoon/test/refactoring/RenameLocalVariableRefactorTest.java

```diff
@@ -0,0 +1,255 @@
+					CtVariable<?> targetVariable = (CtVariable<?>)annotation.getAnnotatedElement();
+					for (String newName : newNames) {
+						boolean renameShouldPass = newName.startsWith("-")==false;
+						if (!renameShouldPass) {
+							newName = newName.substring(1);
+						}
+						if (targetVariable instanceof CtLocalVariable<?>) {
+							//debugging support
+							if(DEBUG.length==3 && DEBUG[1].equals(targetVariable.getSimpleName()) && DEBUG[2].equals(newName)) {
+								//put breakpoint here and continue debugging of the buggy case
+								this.getClass();
+							}
+							checkLocalVariableRename((CtLocalVariable<?>) targetVariable, newName, renameShouldPass);
+						} else {
+							//TODO test rename of other variables, e.g. parameters and catch... later
+						}
+					}
+				});
+		});
+	}
+	
+	protected void checkLocalVariableRename(CtLocalVariable<?> targetVariable, String newName, boolean renameShouldPass) {
+		
+		String originName = targetVariable.getSimpleName();
+		RenameLocalVariableRefactor refactor = new RenameLocalVariableRefactor();
+		refactor.setTarget(targetVariable);
+		refactor.setNewName(newName);
+		if(renameShouldPass) {
+			try {
+				refactor.refactor();
+			} catch(SpoonException e) {
+				throw new AssertionError(getParentMethodName(targetVariable)+" Rename of \""+originName+"\" should NOT fail when trying rename to \""+newName+"\"\n"+targetVariable.toString(), e);
+			}
+			assertEquals(getParentMethodName(targetVariable)+" Rename of \""+originName+"\" to \""+newName+"\" passed, but the name of variable was not changed", newName, targetVariable.getSimpleName());
+			printModelAndTestConsistency(getParentMethodName(targetVariable)+" Rename of \""+originName+"\" to \""+newName+"\"");
+		} else {
+			try {
+				refactor.refactor();
+				fail(getParentMethodName(targetVariable)+" Rename of \""+originName+"\" should fail when trying rename to \""+newName+"\"");
+			} catch(SpoonException e) {
+			}
+			assertEquals(getParentMethodName(targetVariable)+" Rename of \""+originName+"\" failed when trying rename to \""+newName+"\" but the name of variable should not be changed", originName, targetVariable.getSimpleName());
+		}
+		if(renameShouldPass) {
+			rollback(targetVariable, originName);
+		}
+		assertEquals(originName, targetVariable.getSimpleName());
+	}
+	
+	private void rollback(CtLocalVariable<?> targetVariable, String originName) {
+		String newName = targetVariable.getSimpleName();
+		RenameLocalVariableRefactor refactor = new RenameLocalVariableRefactor();
+		refactor.setTarget(targetVariable);
+		//rollback changes
+		refactor.setNewName(originName);
+		try {
+			refactor.refactor();
+		} catch(SpoonException e) {
+			throw new AssertionError(getParentMethodName(targetVariable)+" Rename of \""+originName+"\" to \""+newName+"\" passed, but rename back to \""+originName+"\" failed", e);
+		}
+	}
+
+	private void printModelAndTestConsistency(String refactoringDescription) {

```

Comment:
is an assert-like method, it should be named accordingly `assert*`, eg `assertCorrectModel`